packages/db/migrations/0119.do.add_defaults_to_filters.sql

```diff
@@ -0,0 +1,59 @@
+-- Type: DO
+-- Name: add_defaults_to_filters
+-- Description: Add Defaults to Filters
+
+BEGIN;
+
+ALTER TABLE omnivore.filters
+    ADD COLUMN default_filter boolean NOT NULL DEFAULT false,
+    ADD COLUMN visible boolean NOT NULL DEFAULT true;
+
+CREATE OR REPLACE FUNCTION update_filter_position()
+    RETURNS TRIGGER AS $$
+DECLARE
+    new_position INTEGER;
+BEGIN
+    IF (TG_OP = 'DELETE') THEN
+        UPDATE omnivore.filters SET position = position - 1 WHERE user_id = OLD.user_id AND position > OLD.position;
+        RETURN OLD;
+    ELSIF (TG_OP = 'INSERT' and NEW.position is null) THEN
+        SELECT COALESCE(MAX(position), 0) + 1 INTO new_position FROM omnivore.filters WHERE user_id = NEW.user_id AND name < NEW.name;
+        UPDATE omnivore.filters SET position = position + 1 WHERE user_id = NEW.user_id AND position >= new_position;
+        NEW.position = new_position;
+        RETURN NEW;
+    END IF;
+    RETURN NEW;
+END;
+$$ LANGUAGE 'plpgsql';
+
+INSERT INTO omnivore.filters (user_id, category, name,  filter, position, default_filter)
```

Comment:
Hey, so does this one do an insert for all users? I was thinking of handling that in the resolver just returning default values, but this is probably easier.

Reply:
I think your approach is right, its quite complex to handle the null case I think when you consider partially set default searches.